source/AI.h

```diff
@@ -89,7 +89,7 @@ template <class Type>
 	static void CircleAround(Ship &ship, Command &command, const Ship &target);
 	static void Swarm(Ship &ship, Command &command, const Ship &target);
 	static void KeepStation(Ship &ship, Command &command, const Ship &target);
-	static void Attack(Ship &ship, Command &command, const Ship &target);
+	const void Attack(Ship &ship, Command &command, const Ship &target) const;
```

Comment:
don't forget to remove the first `const` here

Reply:
did the other one as ya suggested remove this or the other

Reply:
You'll still need to remove the first `const` here - otherwise, there will be compile errors as the implementation isn't `const`.

Reply:
The first const here refers to the return type. Returning `const void` is nonsense. The second const indicates that this member function does not modify variables that belong to the AI class (which is true).

Reply:
so remove the first const here
should I move this line above void DoSwarming? since it's just a void then